connector/src/main/java/org/geysermc/connector/utils/SettingsUtils.java

```diff
@@ -121,13 +134,23 @@ public static boolean handleSettingsForm(GeyserSession session, String response)
         }
         int offset = 0;
 
-        // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
-        if (!session.isReducedDebugInfo() && session.getConnector().getConfig().isShowCoordinates()) {
+        if (session.getPreferencesCache().isAllowShowCoordinates() || CooldownUtils.getDefaultShowCooldown() != CooldownUtils.CooldownType.DISABLED) {
             offset++; // Client settings title
 
-            session.getWorldCache().setPrefersShowCoordinates(settingsResponse.getToggleResponses().get(offset));
-            session.getWorldCache().updateShowCoordinates();
-            offset++;
+            // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
+            if (session.getPreferencesCache().isAllowShowCoordinates()) {
+                session.getPreferencesCache().setPrefersShowCoordinates(settingsResponse.getToggleResponses().get(offset));
+                session.getPreferencesCache().updateShowCoordinates();
+                offset++;
+            }
+
+            if (CooldownUtils.getDefaultShowCooldown() != CooldownUtils.CooldownType.DISABLED) {
+                CooldownUtils.CooldownType cooldownType = CooldownUtils.CooldownType.values()[settingsResponse.getDropdownResponses().get(offset).getElementID()];
```

Comment:
Use CooldownType.VALUES here. `values()` is a function that allocates a new array (which we don't need when we have one.

Reply:
thanks, I wasn't sure if there was a significant difference

Reply:
Should the same apply to the gamemode and difficulty stuff below this?

Reply:
I do not believe those have a VALUES static field.

Reply:
the gamerule enum is ours, but the VALUES static field is private. the difficulty enum is mcprotocollibs and doesn't have one.

not sure if this is going out of the scope of this pr